_pytest/assertion/__init__.py

```diff
@@ -113,7 +115,8 @@ def callbinrepr(op, left, right):
         for new_expl in hook_result:
             if new_expl:
                 if (sum(len(p) for p in new_expl[1:]) > 80*8
-                        and item.config.option.verbose < 2):
+                        and item.config.option.verbose < 2
+                        and 'CI' not in os.environ):
```

Comment:
Seems like Jenkins/Hudson indeed don't set `CI`, but both set `BUILD_ID` instead. I'll add that.